library/nameops.ml

```diff
@@ -153,6 +153,9 @@ let name_max na1 na2 =
   | Name _ -> na1
   | Anonymous -> na2
 
+let rec pr_name_list ns =
+  Pp.prlist_with_sep spc (fun v -> Option.default (Pp.str "_") (Option.map pr_id v)) ns
+
```

Comment:
It does seem like `VFun` and `TacFun` should take a `Name.t`. This is a fairly invasive change, and is on its own larger than this pull request.

I did finish making the change, and submitted it as PR #432. I'd like to request that these two changes be considered independently. If we finish the cleanup in #432, then it's easy enough to fix this PR to use `pr_enum pr_name` rather than `pr_id_opt_list`.